### ISSUE TITLE
Automated mode

### ISSUE THREAD

**Issue by dbalabka** [visioncortex/vtracer#7]
Thanks for such a needed and great project! During my exploration, I've tried to achieve similar quality to [VectorMagic](https://vectormagic.com/) automated mode by adjusting different input parameters in https://www.visioncortex.org/vtracer/. Unfortunately, I did succeed. Also, manual parameters adjustments are very time-consuming.

In terms of quality I mean:
1. Less complicated shapes
2. More robustness on image noise

So, I had a thought that it would be great to have an automated mode. Probably, it would be possible to optimize parameters by employe some optimization frameworks (e.g. Hyperopt, Optuna) and optimize some quality function.
Another assumption is that it might be a limitation of the VTrace algorithm. 

If it is possible could you please give me a hint on where to dig to improve output SVG quality? I would love to contribute to your project to improve the quality of output SVG.

**Comment by tyt2y3** [comment]
I think you didn't succeed :(
Have you looked at [visioncortex/visionmagic](https://github.com/visioncortex/visionmagic) ?
It indeed addressed the two factors you mentioned: shape complexity and image complexity.

**Comment by dbalabka** [comment]
@tyt2y3, I've tried to use visioncortex/visionmagic. Unfortunately, it leads to similar results or even worse.

I'm looking for a tool that will accurately convert the raster version of SVG back to SVG. If we will take a look at an example in VTrace Demo, we will see a lot of artifacts:
![image](https://user-images.githubusercontent.com/1269372/132757402-3d36a2c9-1b68-4a47-b441-8cf28fff03e3.png)
![image](https://user-images.githubusercontent.com/1269372/132757602-7817c5cc-53e3-4fb4-980c-95915a89be50.png)

I tried to play around with parameters, but unfortunately, it does not lead to any quality improvements. Comparing with  VectorMagic, it outputs almost perfect SVG.

**Comment by TorRient** [comment]
Hi @dbalabka, do you have another solution like Vectormagic?

**Comment by dbalabka** [comment]
@TorRient probably, [Vtracer](https://github.com/visioncortex/vtracer) is the best alternative. I didn't find anything better.

**Comment by dheerajpg** [comment]
Try this https://vectorizer.ai/ it's free currently as it's in Beta. I hope there will be some open source alternative to this in future

**Comment by tyt2y3** [comment]
A machine learning based pipeline requires a lot of resource to setup, maintain and improved upon. And so for economic reasons, I think it will be difficult for an open source organization to thrive in this space. May be there could be another YOLO, or you will still find classicists (like us) working on algorithms by hand, which requires minimal computational power.

**Comment by dheerajpg** [comment]
is it possible to implement such thing that allows users for you to use their resource to enhance the model, like for example you guys allowing conversion to vector with only credits so users could allow their resources for certain minutes in return for credits that way you could get improve the model.
is it possible with this or you have to completely re-write the codebase if such a thing were to be introduced

sorry this might be stupid idea or question but I'm just asking out of curiosity

**Comment by tyt2y3** [comment]
Yeah, may be a 'shared economy' model can be a viable way to sustain free-as-in-speech open source development.
Or honestly, I don't have a definitive idea either. I am glad that we had such a conversation though.
